theano/gpuarray/opt.py

```diff
@@ -738,16 +736,19 @@ def local_gpuajoin_1(node):
 @register_opt('fast_compile')
 @op_lifter([tensor.Split])
 @register_opt2([tensor.Split], 'fast_compile')
-def local_gpua_split(node, context_name):
+def local_gpua_split(op, context_name, inputs):
 #TODO use props
-    return GpuSplit(node.op.len_splits)
+    return GpuSplit(op.len_splits)
 
 
 @register_opt('fast_compile')
 @op_lifter([tensor.Subtensor])
 @register_opt2([tensor.Subtensor], 'fast_compile')
-def local_gpua_subtensor(node, context_name):
-    x = node.inputs[0]
+def local_gpua_subtensor(op, context_name, inputs):
+    x = inputs[0]
+    node = op.make_node(*inputs)
+    fg = gof.FunctionGraph(inputs, node.outputs)
+    node = fg.toposort()[0]
```

Comment:
The right way to keep the complicated condition, is to pass clients too.
In fact, it is a list of list of clients. like
```
c = []
for o in node.outputs:
   c.append(o.clients)
```